### PR TITLE
Added call-service type and extended attr

### PR DESCRIPTION
**Pull request by emericklaw**
I was going to create this feature and saw @dklemm had forked already.

I am creating this pull request on his behalf, hopefull that is OK!

A separate pull request from me will have the updated documentation.

### CODE REVIEW

**Comment by emericklaw** [comment]
This closes #12

File: card.js [hunk 1]
```diff
@@ -93,8 +101,13 @@ class RGBLightCard extends HTMLElement {
     }
 
     applyColor(color) {
-        const serviceData = { entity_id: this.config.entity, ...color, icon_color: undefined, type: undefined };
-        this._hass.callService(color.type || 'light', 'turn_on', serviceData);
+        if (color.type === 'call-service') {
+            const [domain, service] = color.service.split('.');
+            this._hass.callService(domain, service, color.service_data);
+        } else {
+            const serviceData =  { entity_id: this.config.entity, ...color, icon_color: undefined, type: undefined };
```
Comment (emericklaw) [inline card.js]: Extra space needs removing between the = and {
```suggestion
            const serviceData = { entity_id: this.config.entity, ...color, icon_color: undefined, type: undefined };
```

**Comment by dklemm** [comment]
Great, I was just waiting to update the document before submitting myself but got sidetracked

File: card.js (hunk 1 above)
Comment (bokub) [reply card.js]: Actually, you just need to run `npm run lint` to fix style, and commit again

**Comment by bokub** [comment]
I'm merging the PR and doing a little bit of rework

File: card.js [hunk 2]
```diff
@@ -68,17 +68,25 @@ class RGBLightCard extends HTMLElement {
...
             if (type === 'light' && !config.entity && !color.entity_id) {
                 throw new Error(`You need to define entity or colors[${c}].entity_id`);
             }
             // If scene or script, ensure entity_id is defined
-            if (type !== 'light' && !color.entity_id) {
+            if (['script', 'scene'].indexOf(type) > -1 && !color.entity_id) {
                 throw new Error(`You need to define colors[${c}].entity_id`);
             }
+            // If call-service, ensure service is defined
+            if ('type' === 'call-service' && !color.service) {
```
Comment (bokub) [inline card.js]: `type !== 'type'`

**Pull request merged**